File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
       vector<int> s;
        vector<string> res;
        unordered_set<int> set(target.begin(), target.end());
        for (int i = 1; i <= n; i++) {
            s.push_back(i);
            res.push_back("Push");
            if (set.find(s.back()) == set.end()) {
                s.pop_back();
                res.push_back("Pop");
            }
            if (target == s) {
                return res;
            }
        }
        return res; 
        }
};

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        ans = []
        i = 1
        
        for num in target:
            while i < num :
                ans.append("Push")
                ans.append("Pop")
                i += 1

            ans.append("Push")
            i += 1

        return ans

